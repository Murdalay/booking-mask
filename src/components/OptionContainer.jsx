import React from 'react';
import styled from 'styled-components';
import { Container, utils } from 'styled-minimal';

const { spacer } = utils;

const OptionContainer = styled(Container)`
  height: ${props => props.theme.sizes.controlHeight}px;
  display: flex;
  padding: ${spacer(1)};
  align-content: center;
  align-items: center;
  border-bottom: 1px solid #999;
  pointer-events: auto !important;
  input {
    height: ${props => props.theme.sizes.controlHeight - 20}px;
    font-size: 1.2rem !important;
    width: 80px;
    &:focus {
      outline: none;
    }
  }
`;

export default React.memo(OptionContainer);
