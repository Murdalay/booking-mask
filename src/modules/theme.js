export const rowHeight = 70;
export const controlHeight = 55;
export const appColor = '#00b4d5';
export const secondaryColor = '#88e4f5';
export const actionColor = '#66b4e5';
export const inputMinWidth = 150;
export const defaultIconColor = '#444';
export const defaultTextColor = '#555';
export const borderColor = 'rgb(204, 204, 204)';
export const easing = 'cubic-bezier(0.35, 0.01, 0.77, 0.34);';
export const inputFontSize = '1.5rem';
export const inputBackgroundColor = '#fff';

export default {
  breakpoints: [400, 768, 1024, 1280, 1920],
  palette: {
    primary: appColor,
    secondary: secondaryColor,
    actionColor,
    appColor,
    borderColor,
    defaultTextColor,
    defaultIconColor,
    inputBackgroundColor,
  },
  sizes: {
    rowHeight,
    controlHeight,
    borderColor,
    inputMinWidth,
    inputFontSize,
  },
  input: {
    borderRadius: {
      xs: 0,
      lg: 0,
      xl: 0,
      default: 0,
    },
  },
  button: {
    borderRadius: {
      xs: 0,
      lg: 0,
      xl: 0,
    },
    padding: {
      lg: [12, 28],
      xl: [14, 32],
    },
  },
};
