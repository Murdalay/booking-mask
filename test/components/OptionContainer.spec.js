import React from 'react';
import theme from 'modules/theme';

import OptionContainer from 'components/OptionContainer';

describe('OptionContainer', () => {
  const wrapper = mount(<OptionContainer theme={theme} />);

  it('should render properly', () => {
    expect(wrapper).toMatchSnapshot();
  });
});
