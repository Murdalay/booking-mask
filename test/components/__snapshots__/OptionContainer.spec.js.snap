// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`OptionContainer should render properly 1`] = `
.c0 {
  box-sizing: border-box;
  margin-left: auto;
  margin-right: auto;
  padding-left: 16px;
  padding-right: 16px;
  max-width: 1440px;
  position: relative;
  width: 100%;
  height: 55px;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  padding: 4px;
  -webkit-align-content: center;
  -ms-flex-line-pack: center;
  align-content: center;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  border-bottom: 1px solid #999;
  pointer-events: auto !important;
}

.c0 input {
  height: 35px;
  font-size: 1.2rem !important;
  width: 80px;
}

.c0 input:focus {
  outline: none;
}

@media (min-width:768px) {
  .c0 {
    padding-left: 32px;
    padding-right: 32px;
  }
}

<OptionContainer
  theme={
    Object {
      "breakpoints": Array [
        400,
        768,
        1024,
        1280,
        1920,
      ],
      "button": Object {
        "borderRadius": Object {
          "lg": 0,
          "xl": 0,
          "xs": 0,
        },
        "padding": Object {
          "lg": Array [
            12,
            28,
          ],
          "xl": Array [
            14,
            32,
          ],
        },
      },
      "input": Object {
        "borderRadius": Object {
          "default": 0,
          "lg": 0,
          "xl": 0,
          "xs": 0,
        },
      },
      "palette": Object {
        "actionColor": "#66b4e5",
        "appColor": "#00b4d5",
        "borderColor": "rgb(204, 204, 204)",
        "defaultIconColor": "#444",
        "defaultTextColor": "#555",
        "inputBackgroundColor": "#fff",
        "primary": "#00b4d5",
        "secondary": "#88e4f5",
      },
      "sizes": Object {
        "borderColor": "rgb(204, 204, 204)",
        "controlHeight": 55,
        "inputFontSize": "1.5rem",
        "inputMinWidth": 150,
        "rowHeight": 70,
      },
    }
  }
  verticalPadding={false}
>
  <div
    className="c0"
  />
</OptionContainer>
`;
